src/mono/wasm/Wasm.Build.Tests/Templates/WasmTemplateTests.cs

```diff
@@ -491,5 +491,68 @@ public void BuildAndRunForDifferentOutputPaths(string config, bool appendRID, bo
+            string projectDirectory = Path.GetDirectoryName(projectFile)!;
+            bool aot = true;
+
+            UpdateProgramCS();
+            UpdateConsoleMainJs();
+
+            AddItemsPropertiesToProject(projectFile, "<RunAOTCompilation>true</RunAOTCompilation>");
+            AddItemsPropertiesToProject(projectFile, "<WasmStripILAfterAOT>true</WasmStripILAfterAOT>");
+            AddItemsPropertiesToProject(projectFile, "<WasmEnableWebcil>false</WasmEnableWebcil>");
+
+            var buildArgs = new BuildArgs(projectName, "Release", aot, id, null);
+            buildArgs = ExpandBuildArgs(buildArgs);
+
+            BuildTemplateProject(buildArgs,
+                        id: id,
+                        new BuildProjectOptions(
+                            CreateProject: false,
+                            HasV8Script: false,
+                            MainJS: "main.mjs",
+                            Publish: true,
+                            TargetFramework: BuildTestBase.DefaultTargetFramework,
+                            UseCache: false,
+                            IsBrowserProject: false,
+                            AssertAppBundle: false));
+
+            string runArgs = $"run --no-silent --no-build -c Release";
+            runArgs += " x y z";
+            var res = new RunCommand(s_buildEnv, _testOutput, label: id)
+                                .WithWorkingDirectory(_projectDir!)
+                                .ExecuteWithCapturedOutput(runArgs)
+                                .EnsureExitCode(42);
+            
+            string frameworkDir = Path.Combine(projectDirectory, "bin", "Release", BuildTestBase.DefaultTargetFramework, "browser-wasm", "AppBundle", "_framework");
+            string objBuildDir = Path.Combine(projectDirectory, "obj", "Release", BuildTestBase.DefaultTargetFramework, "browser-wasm", "wasm", "for-publish");
+            string origAssemblyDir = Path.Combine(objBuildDir, "aot-in");
+            string strippedAssemblyDir = Path.Combine(objBuildDir, "stripped");
+            Assert.True(Directory.Exists(origAssemblyDir), $"Could not find the original AOT input assemblies dir: {origAssemblyDir}");
+            Assert.True(Directory.Exists(strippedAssemblyDir), $"Could not find the stripped assemblies dir: {strippedAssemblyDir}");
+
+            string assemblyToExam = "System.Private.CoreLib.dll";
+            string originalAssembly = Path.Combine(objBuildDir, origAssemblyDir, assemblyToExam);
+            string strippedAssembly = Path.Combine(objBuildDir, strippedAssemblyDir, assemblyToExam);
+            string bundledAssembly = Path.Combine(frameworkDir, assemblyToExam);
+            Assert.True(File.Exists(originalAssembly));
+            Assert.True(File.Exists(strippedAssembly));
+            Assert.True(File.Exists(bundledAssembly));
+
+            string compressedOriginalAssembly = Compress(originalAssembly);
```

Comment:
Why do we need to compare the compressed sizes?

Reply:
Because the .dll files are the same size. The trim was done by replacing method body by zeros.